pkg/queue/health/probe.go

```diff
@@ -202,3 +219,69 @@ func isHTTPProbeReady(res *http.Response) bool {
 	// response status code between 200-399 indicates success
 	return res.StatusCode >= 200 && res.StatusCode < 400
 }
+
+// GRPCProbe checks that gRPC connection can be established to the address.
+func GRPCProbe(config GRPCProbeConfigOptions) error {
+	dialer := &net.Dialer{
+		Control: func(network, address string, c syscall.RawConn) error {
+			return c.Control(func(fd uintptr) {
+				unix.SetsockoptLinger(int(fd), syscall.SOL_SOCKET, syscall.SO_LINGER, &unix.Linger{Onoff: 1, Linger: 1})
+			})
+		},
+	}
```

Comment:
Can you reference where you got this from

https://github.com/kubernetes/kubernetes/blob/master/pkg/probe/dialer_others.go#L33

Reply:
and mention why we are using these options

Reply:
I have a question about it.

Is it necessary to respond to Windows container? 🤔

Reply:
No - we currently don't have windows support